src/charm.py

```diff
@@ -487,6 +488,14 @@ def _on_start(self, event: ops.charm.StartEvent):
         # available and the status was stuck on "Waiting for IP address". Adding this hook as a workaround.
         self._common_exit_hook()
 
+    def _on_install(self, _):
+        """Event handler for the install event during which we will update the K8s service"""
+        # At this point, it's debatable whether we should put the charm into a blocked state.
```

Comment:
The "common exit hook" is really an anti-pattern I'd like to avoid (to be discussed in charm feedback sessions), precisely because of this sort of situation. I don't think this should be a case for `BlockedState`, but it should definitely log that something is up